compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -898,21 +898,43 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
 
       canWidenAbstract && acc(true, tp)
 
-    def tryBaseType(cls2: Symbol) = {
-      val base = nonExprBaseType(tp1, cls2).boxedIfTypeParam(tp1.typeSymbol)
+    def tryBaseType(cls2: Symbol) =
+
+      def computeBase(tp: Type): Type = tp.widenDealias match
+        case tp @ AndType(tp1, tp2) =>
```

Comment:
I would call the operands `tp11 and tp12`  to avoid shadowing the existing tp1/tp2.